backend/app/services/dane.py

```
@@ -137,22 +179,147 @@ def parse_tlsa_record(record: str, *, query_name: str, mx_host: str) -> TLSAReco
+    try:
+        from cryptography import x509
+        from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+    except ImportError as exc:  # pragma: no cover - dependency is present in packaged builds
+        return TLSASuggestion(
+            query_name=query_name,
+            mx_host=mx_host,
+            error=f"cryptography dependency is unavailable: {exc}",
+        )
+
+    try:
+        context = ssl._create_unverified_context()  # nosec B323 - read-only certificate fetch.
```

Ruff (0.15.20)

[error] Python allows using an insecure context via the `_create_unverified_context` that reverts to the previous behavior that does not validate certificates or perform hostname checks.

(S323)